openssl/src/cms.rs

```diff
@@ -80,4 +81,51 @@ impl CmsContentInfo {
             Ok(CmsContentInfo::from_ptr(cms))
         }
     }
+
+    /// Given a signing cert `signcert`, private key `pkey`, an optional certificate stack `certs`,
+    /// data `data` and flags `flags`, create a CmsContentInfo struct.
+    ///
+    /// OpenSSL documentation at [`CMS_sign`]
+    ///
+    /// [`CMS_sign`]: https://www.openssl.org/docs/manmaster/man3/CMS_sign.html
+    pub fn sign<T: HasPrivate>(
+        signcert: &X509,
+        pkey: &PKeyRef<T>,
+        certs: Option<&Stack<X509>>,
+        data: &[u8],
+        flags: u32,
```

Comment:
This should be a special type like `SslOptions` is now.

Reply:
I'll address this in another PR (also containing `CMS_add1_signer` and friends).